src/coprocessor/dag/expr/builtin_string.rs

```diff
@@ -77,6 +78,28 @@ impl ScalarFunc {
             str::from_utf8(&s)?.to_lowercase().into_bytes(),
         )))
     }
+
+    #[inline]
+    pub fn un_hex<'a, 'b: 'a>(
+        &'b self,
+        ctx: &mut EvalContext,
+        row: &'a [Datum],
+    ) -> Result<Option<Cow<'a, [u8]>>> {
+        let s = try_opt!(self.children[0].eval_string(ctx, row));
+        let hex_string = if s.len() % 2 == 1 {
+            let mut vec = vec![b'0'];
+            vec.extend_from_slice(&s);
+            vec
+        } else {
+            s.to_vec()
+        };
+        let result = Vec::from_hex(hex_string);
+        result
+            .map_err(|_e| {
+                Error::wrong_value_for_type("string", str::from_utf8(s.borrow()).unwrap(), "unhex")
```

Comment:
It seems we should return null for the following case https://dev.mysql.com/doc/refman/5.7/en/string-functions.html#function_unhex
```
mysql> SELECT UNHEX('GG');
+-------------+
| UNHEX('GG') |
+-------------+
| NULL        |
+-------------+
```

Reply:
Yes, mysql also returns a warning. So we skip the warning and just return NULL, right?

Reply:
I think we can keep compatible with `tidb` and return a Null.